### DISCUSSION TITLE
Deciding on front-end

### DISCUSSION THREAD

**Discussion by nicolecarman** [Nerds-Who-Code/Mental-Health-Tracker#9] [General]
Please answer the following questions in a comment on this post so that we can determine how we will handle the front-end, whether we code it from scratch or use a framework. This is required by all team members. Please be realistic with your current knowledge and expertise, and with what you are willing to learn specifically for this project when answering these questions.
_______________________________________

1. Are you interested in front-end, back-end, or both? It's okay if you are interested in one but have no prior knowledge with it yet. We are doing this to learn.

2. What markup and programming languages do you know right now? Please describe your proficiencies with each one you list, whether you're a beginner, intermediate, etc...

3. If you are interested in front-end, would you rather help code from scratch (a literal blank file), or would you like to consider working with React (front-end framework, not coding from scratch)? It's okay to be interested in both or to be unsure, just say so.

4. We are discussing the following for this project: HTML, CSS, Javascript, React, Express.js, and MongoDB. If you ARE NOT familiar with any of these, are there any that you are interested in learning for this project? It's okay to say no!

It is okay if you aren't interested in learning a new language or framework. Just be sure you tell us that. We need to get an idea of what everyone knows, as well as what everyone is willing to learn, so that we can determine how to proceed and start coding.

Quoting CutePeanut:
If you have experience only in front-end, but you want to learn back-end you can mention this as well so you can partner with someone more knowledgeable in the back-end for learning.

**Comment by nicolecarman** [comment]
1. I am interested primarily in back-end. However, I will be available to assist with front-end as needed.

2. I have proficient/intermediate knowledge in HTML and CSS. I feel comfortable writing from scratch. I started learning Javascript last month and I know the basics, but not data structures. I'm a beginner with C++ (I know we aren't using it).

3. I would rather get my feet wet with React than write HTML and CSS from scratch.

4. I'm not well-versed in Javascript, and I don't know React, Express.js, or MongoDB, but I'm open to learning and using all of them for this project.

**Comment by xintamosaik** [comment]
1. would love to do the full-stack
2. I am quite good with the vanilla stack (HTML/CSS/JS), good with LAMP (Linux Apache MariaDB PHP) and ok with Docker. 
3./4. I would like to learn EVERYTHING, but I have to be realistic. I will probably have a good chance in understanding node.js or express and I might be helpful in connecting a database. I have no idea yet what React is like and what the learning curve is.

**Comment by nab-sa** [comment]
1. I am focusing on front-end only.

2. I'm intermediate(-ish) in vanilla HTML / CSS 

3. I'd like to learn React but my lack of knowledge in JavaScript makes me worry that the learning curve might be quite steep.

4. I have zero knowledge in React, very little in JavaScript.

**Comment by SherlockX9** [comment]
1. I would like to help with both front-end and back-end.
2. I am intermediate(ish) with HTML and CSS.
3. I know JavaScript at a basic level.
4. Would like to learn react.
5. For backend, I am ok with php and MySQL, but I don't know Mongo DB or Express JS. However I would really like to learn them and I don't think it will be difficult to pick up on it.

**Comment by erikgalajda** [comment]
1. interested in both, have experience in both
2. HTML, CSS, JS (react, express, node), python, SQL, DB/server/container knowledge -> higher intermediate level
3. wanna advance my react skills
4. i have never used mongodb so that would be interesting for me

**Comment by med4kat** [comment]
1. I’m interested in front-end, but would be glad to learn something new about React, MongoDB or other stuff.
2. Hmtl/Css  - intermediate(ish);  SVG animation - basics; Python - basics; SQL - basics; 
3. I can create html/css and one can refactor the code into React components as one sees fit. I had an experience working this way. It went very well. 
4. I’m interested in learning all of those. But most of all I’m interested in working in a team, do some marketing, practice agile methodology (scrum), create a comfortable and efficient environment for the team, improve my soft skills, build useful usable open source app to help people. So I can easily step out of coding and do other stuff if required.

**Comment by steph-koopmanschap** [comment]
1. Interested mostly in Back-end, but also a lil bit of Front-End. 

2. Good understanding of HTML/CSS/JS and NodeJs
basic understanding of React, Vue, and Nxinx. 

3. I like to learn more about a React or Vue. So I prefer those.

4. I wanna learn more Back-end stuff. ExpressJS, SQL, MongoDB, CRUD API, etc
Server maintenance and Docker.
